src/libraries/System.Diagnostics.DiagnosticSource/src/System.Diagnostics.DiagnosticSource.csproj

```diff
@@ -7,7 +7,7 @@
     <EnableTrimAnalyzer Condition="$([MSBuild]::GetTargetFrameworkIdentifier('$(TargetFramework)')) == '.NETFramework'">false</EnableTrimAnalyzer>
     <IsPackable>true</IsPackable>
     <!-- Lifetime rules introduced in C# 11 impact scenarios in net6 framework -->
-    <LangVersion Condition="'$(TargetFramework)' == 'net6.0'">10</LangVersion>
+    <LangVersion Condition="'$(TargetFramework)' == 'net6.0' or '$(TargetFramework)' == 'net7.0'">10</LangVersion>
```

Comment:
CC @AaronRobinsonMSFT as he is the one introduced this.

Reply:
This was because of the `ref` field support and the project using the .NET 7 Preview ref assemblies instead of the live build assemblies. Once the SDK for the repo was updated, this can likely be removed and the code fixed up a bit.

/cc @jakobbotsch

Reply:
Based on Aaron's reply, this setting shouldn't be needed anymore:

```suggestion
```

Reply:
This change should also be forward ported into main.